src/Path.hs

```diff
@@ -110,6 +110,7 @@ data PathParseException
   | InvalidAbsFile FilePath
   | InvalidRelFile FilePath
   | Couldn'tStripPrefixDir FilePath FilePath
+  | NoParent
```

Comment:
I'd pop the offending `FilePath` in there, this has been useful in Stack where an exception is thrown of an offending filename and you get a hint.